gobblin-runtime/src/main/java/gobblin/runtime/AbstractJobLauncher.java

```diff
@@ -245,18 +247,20 @@ public void apply(JobListener jobListener, JobContext jobContext)
     final List<WorkUnit> skippedWorkUnits = new ArrayList<>();
     for (WorkUnit workUnit : workUnits.get()) {
       if (workUnit instanceof MultiWorkUnit) {
```

Comment:
1) At multiple places, list of WorkUnits will be passed and invoked. It's essential to remove those WorkUnits from the list.
2) In order to support skipping of some WorkUnit inside of a MultiWorkUnits, this is essential.